runtime/cudaq/platform/mqpu/helpers/MQPUUtils.cpp

```diff
@@ -79,6 +79,26 @@ cudaq::AutoLaunchRestServerProcess::AutoLaunchRestServerProcess(
   if (!serverApp)
     throw std::runtime_error("Unable to find CUDA-Q REST server to launch.");
 
+  // If the CUDAQ_DYNLIBS env var is set (typically from the Python
+  // environment), use that data to preload the specified .so files.
+  std::string preload;
+  std::optional<llvm::SmallVector<llvm::StringRef>> Env;
+  if (auto *ch = getenv("CUDAQ_DYNLIBS")) {
+    Env = llvm::SmallVector<llvm::StringRef>();
+    if (auto *p = getenv("LD_PRELOAD")) {
+      preload = p;
+      if (preload.size() > 0)
+        preload += ":";
+    }
+    preload += std::string(ch);
+    for (char **env = environ; *env != nullptr; ++env) {
+      if (!std::string(*env).starts_with("LD_PRELOAD="))
+        Env->push_back(*env);
+    }
+    preload = "LD_PRELOAD=" + preload;
```

Comment:
The issue with using LD_PRELOAD is that it fails if the libraries can't be loaded. 
Currently, we don't have a CUDA-Q build shipped separately without GPU components. Hence, LD_PRELOAD makes it impossible to use CUDA-Q without GPU/a CUDA installation right now. We either need to ship a version without CUDA (and generally I prefer not adding more packages), or we need a non-failing alternative.

Reply:
What was the reason to use LD_PRELOAD instead of just adding these to the LD_LIBRARY_PATH like we do in the *qpud.py?

Reply:
I took the liberty of pushing two commits that should have the (almost) exact same logic as we have in the python script for qpud.